src/app.py

```diff
@@ -0,0 +1,80 @@
+import gradio as gr
+import torch
+import os
+import yaml
+import cv2
+
+from yaml import FullLoader
+from detectron2.engine import DefaultPredictor
+from detectron2.config import get_cfg
+from detectron2 import model_zoo
+from detectron2.utils.visualizer import Visualizer
+from detectron2.data import MetadataCatalog, DatasetCatalog
+
+from dataset_utils import get_reindeer_dicts
+
+# function to load the model
+def load_model():
+    dataset_name = "reindeer_test"
+    
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    config_path = os.path.join(current_dir, './../config.yaml')
+    
+    with open(config_path) as f:
+        cfgP = yaml.load(f, Loader=FullLoader)
+    
+    img_dir = cfgP["TILE_TEST_FOLDER_PATH"]
+    annotations_file = cfgP["TILE_TEST_ANNOTATION_PATH"]
+    model_weights = cfgP["MODEL_WEIGHTS"]
```

Comment:
You can check Hydra/OmegaConf and django-environ (which does not require Django actually).

Reply:
Will do this for other projects moving forward 👯